manifests/configuration.pp

```diff
@@ -0,0 +1,110 @@
+# Manage sysctl variable values.
+#
+# Parameters:
+#  $value:
+#    The value for the sysctl parameter. Mandatory, unless $ensure is 'absent'.
+#  $prefix:
+#    Optional prefix for the sysctl.d file to be created. Default: none.
+#  $ensure:
+#    Whether the variable's value should be 'present' or 'absent'.
+#    Defaults to 'present'.
+#
+# Sample Usage :
+#  sysctl::configuration { 'net.ipv6.bindv6only': value => '1' }
+define sysctl::configuration (
+  String $variable               = $title,
+  String $ensure                 = 'present',
+  Variant[String,Undef] $value   = undef,
+  Variant[String,Undef] $prefix  = undef,
+  String $suffix                 = '.conf',
+  Variant[String,Undef] $comment = undef,
+  Variant[String,Undef] $content = undef,
+  Variant[String,Undef] $source  = undef,
+  Boolean $enforce               = true,
+  String $sysctl_binary          = '/sbin/sysctl',
+  String $sysctl_dir_path        = '/etc/sysctl.d',
+) {
+
+  # If we have a prefix, then add the dash to it
+  if $prefix {
+    $_sysctl_d_file = "${prefix}-${variable}${suffix}"
+  } else {
+    $_sysctl_d_file = "${variable}${suffix}"
```

Comment:
thanks for the tip with the periods, worked fine for me without quoting, but now i quoted everything just to don't get in trouble later.

concerning the `variable` & `title` stuff, i am quite sure that my fix is correct. 


for the following hiera code:
```
'test-net.ipv6.conf.default.autoconf': { variable: 'net.ipv6.conf.default.autoconf', value: '0', prefix: '40' }
````

without the fix in the `sysctl.d-file.erb` file i get the following error:

```
==> default: Error: Execution of '/sbin/sysctl -p /etc/sysctl.d/40-net.ipv6.conf.default.autoconf.conf20160922-29874-1qu71uf' returned 255: sysctl: cannot stat /proc/sys/test-net/ipv6/conf/default/autoconf: No such file or directory
==> default: Error: /Stage[main]/Sysctl/Sysctl::Configuration[test-net.ipv6.conf.default.autoconf]/File[/etc/sysctl.d/40-net.ipv6.conf.default.autoconf.conf]/ensure: change from absent to present failed: Execution of '/sbin/sysctl -p /etc/sysctl.d/40-net.ipv6.conf.default.autoconf.conf20160922-29874-1qu71uf' returned 255: sysctl: cannot stat /proc/sys/test-net/ipv6/conf/default/autoconf: No such file or directory
==> default: Notice: /Stage[main]/Sysctl/Sysctl::Configuration[test-net.ipv6.conf.default.autoconf]/Exec[enforce-sysctl-value-net.ipv6.conf.default.autoconf]: Dependency File[/etc/sysctl.d/40-net.ipv6.conf.default.autoconf.conf] has failures: true
==> default: Warning: /Stage[main]/Sysctl/Sysctl::Configuration[test-net.ipv6.conf.default.autoconf]/Exec[enforce-sysctl-value-net.ipv6.conf.default.autoconf]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Sysctl/Sysctl::Configuration[test-net.ipv6.conf.default.autoconf]/Exec[remove-net.ipv6.conf.default.autoconf-from-sysctl.conf]: Dependency File[/etc/sysctl.d/40-net.ipv6.conf.default.autoconf.conf] has failures: true
==> default: Warning: /Stage[main]/Sysctl/Sysctl::Configuration[test-net.ipv6.conf.default.autoconf]/Exec[remove-net.ipv6.conf.default.autoconf-from-sysctl.conf]: Skipping because of failed dependencies
```

with the fix in the `sysctl.d-file.erb` file but without the fix in the `configuration.pp` i get no error but the filename is wrong:

i get `40-net.ipv6.conf.default.autoconf.conf` but would assume `40-test-net.ipv6.conf.default.autoconf.conf`


after the fix i get no error and the correct filename `40-test-net.ipv6.conf.default.autoconf.conf`.